core/modules/user/user.admin.inc

```diff
@@ -628,6 +638,8 @@ function user_admin_roles_add_submit($form, &$form_state) {
   $role = (object) array(
     'name' => $form_state['values']['name'],
     'label' => $form_state['values']['label'],
+    // Add new roles directly after the authenticated role.
+    'weight' => $form_state['values']['roles'][BACKDROP_AUTHENTICATED_ROLE]['weight'] + 1,
```

Comment:
I don't think this will work reliably unless we increment all the other roles by 1 also. I'll test.

Reply:
Yeah, the current weighting behaves strangely. To the untrained eye, new roles are arbitrarily placed somewhere between `Anonymous` and `Administrator`.

Reply:
I know 🤔 ...this is why I said "I've implemented this in a simple way..." in the issue comment. Perhaps this will need to be a follow up if the logic is too complicated.

Reply:
...I'll revert this. I've filed a separate follow-up here: #https://github.com/backdrop/backdrop-issues/issues/4507

Reply:
...reverted 👍